lib/cdo/contact_rollups.rb

```diff
@@ -427,19 +428,24 @@ def self.update_data_from_forms(form_kind)
 
       role = data['role_s']
 
-      # Skip if this form data has no info at all
-      next if street_address.nil? && city.nil? && state.nil? && postal_code.nil? && country.nil? && role.nil?
-
-      # Truncate all fields to 255 chars. We have some data longer than 255 chars but it is all garbage that somebody typed.
-      update_data = {}
-      update_data[:street_address] = street_address[0...255] unless street_address.nil?
-      update_data[:city] = city[0...255] unless city.nil?
-      update_data[:state] = state[0...255] unless state.nil?
-      update_data[:postal_code] = postal_code[0...255] unless postal_code.nil?
-      update_data[:country] = country[0...255] unless country.nil?
+      any_update = false
 
-      # add to batch to update
-      update_batch += conn[DEST_TABLE_NAME.to_sym].where(email: email).update_sql(update_data) + ";" unless update_data.empty?
+      # Skip if this form data has no info at all
+      if street_address.present? || city.present? || state.present? || postal_code.present? || country.present?
+        # If any geo input is present in form, update ALL geo fields including setting NULL values in DB for any that
+        # we don't have from this form. This will clear out any previous geo data from IP geo.
+        # Truncate all fields to 255 chars. We have some data longer than 255 chars but it is all garbage that somebody typed
+        update_data = {}
+        update_data[:street_address] = street_address.present? ? street_address[0...255] : nil
+        update_data[:city] = city.present? ? city[0...255] : nil
+        update_data[:state] = state.present? ? state[0...255] : nil
+        update_data[:postal_code] = postal_code.present? ? postal_code[0...255] : nil
+        update_data[:country] = country.present? ? country[0...255] : nil
```

Comment:
Did the first 2 suggestions, thanks. I considered removing `any_update` but decided to leave it. If I remove it, then I wind up with two copies of the conditional logic `update_data.empty?` and `role.blank?`. If the logic ever changes you somehow need to know to update the logic in two places, and they aren't even on the same screen. I will leave in the `any_update` variable because it is more bug-resistant and future-proof.